e2e/specs/websocket_reconnects.spec.js

```diff
@@ -44,26 +44,51 @@ describe("websocket reconnects", () => {
         }, 100);
       });
 
-      // Wait until we've disconnected.
+      // Wait until we've disconnected and reconnected.
       cy.get("[data-testid='stStatusWidget']").should(
         "have.text",
-        "Connecting"
+        "Connecting",
       );
-      // Wait until we've reconnected and rerun the script.
       cy.get("[data-testid='stStatusWidget']").should("not.exist");
 
-      cy.get(".stMarkdown").contains(`count: ${expectedCount}`);
+      cy.getIndexed(".stMarkdown", 0).contains(`count: ${expectedCount}`);
+      cy.getIndexed(".stMarkdown", 1).contains(`slow operations attempted: 0`);
     }
   });
 
+  it("reruns script when script is interrupted by a websocket disconnect", () => {
+    cy.get(".stCheckbox input").click({ force: true });
+
+    cy.window().then((win) => {
+      setTimeout(() => {
+        win.streamlitDebug.disconnectWebsocket();
+      }, 100);
+    });
+
+    // Wait until we've disconnected and reconnected.
+    cy.get("[data-testid='stStatusWidget']").should("have.text", "Connecting");
+    cy.get("[data-testid='stStatusWidget']").should(
+      "not.have.text",
+      "Connecting",
+    );
+
+    // Counter was incremented once when the checkbox was selected and a second
+    // time on the script rerun. We set a high timeout for this check because
+    // the operation we're simulating is naturally slow to give us time to
+    // disconnect the websocket while it's running.
+    cy.getIndexed(".stMarkdown", 1).contains(`slow operations attempted: 2`, {
+      timeout: 10000,
+    });
+  });
```

Comment:
would it be possible to bite the bullet and migrate to playwright first? I don't think it makes too much sense to extend the cypress tests at this point. Unless you are on a very critical timeline.

Reply:
@vdonato I went ahead and migrated the existing tests in this PR: https://github.com/streamlit/streamlit/pull/9126
If that looks good, could you build on top of it and migrate your new test?

Reply:
Thanks! I'll rebase onto `develop` and build on top of the new test